### PR TITLE
F timeperiods

### PR DESCRIPTION
**Pull request by smbambling**
Define to create timeperiod objects

### CODE REVIEW

**Comment by smbambling** [comment]
Reference https://dev.icinga.org/issues/7217

**Comment by cropalato** [comment]
Hi @smbambling, is it missing some attributes? I finished my own define, if you want you can find in my branch https://github.com/cropalato/puppet-icinga2/tree/7217. As I saw your PR I did't add a new one.

**Comment by smbambling** [comment]
@cropalato: Yep looks like I was missing methods additionally I was hard setting the import template as I believe that is required ( I could be wrong ).  But I have taken your branch code as an example and added methods and a modifiable template to import in the even its needed.

**Comment by cropalato** [comment]
Sorry, but should me nice if you add something on README.md too. =D

**Comment by nickchappell** [comment]
Just for reference, the dev.icinga.org issue is: https://dev.icinga.org/issues/7217

File: README.md [hunk 1]
```diff
@@ -633,6 +634,28 @@ icinga2::object::hostgroup { 'admins':
...
 
+####[`icinga2::object::timeperiod`](id:object_timeperiod)
+
+This defined type creates **TimePeriod** objects
+
+Example usage:
+
+````
+icinga2::object::timeperiod { 'bra-office-hrs':
+  display_name => 'Brazilian WorkTime Hours',
```
Comment (nickchappell) [inline README.md]: This is `timeperiod_display_name` in the params list in the manifest. Was one of them a typo?

File: manifests/object/timeperiod.pp [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+# See the inline comments.
+#
+
+define icinga2::object::timeperiod (
+  $object_timeperiod_name        = $name,
+  $timeperiod_template_to_import = 'legacy-timeperiod',
+  $timeperiod_display_name       = undef,
+  $timeperiod_methods            = undef,
+  $timeperiod_ranges             = {},
+  $timeperiod_target_dir         = '/etc/icinga2/objects/timeperiods',
```
Comment (nickchappell) [inline manifests/object/timeperiod.pp:18]: Would you be able to take off the `timeperiod_` from the `target_dir` and `target_file*` params so they match the other defined types in the module?

File: manifests/object/timeperiod.pp [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+#
+# === Parameters
+#
+# See the inline comments.
+#
+
+define icinga2::object::timeperiod (
+  $object_timeperiod_name        = $name,
+  $timeperiod_template_to_import = 'legacy-timeperiod',
+  $timeperiod_display_name       = undef,
```
Comment (nickchappell) [inline manifests/object/timeperiod.pp:15]: Would you be able to remove the `timeperiod_` from this parameter name to match the other defined types' params lists?

File: manifests/object/timeperiod.pp [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  $timeperiod_methods            = undef,
+  $timeperiod_ranges             = {},
```
Comment (nickchappell) [inline manifests/object/timeperiod.pp]: Same with `timeperiod_methods` and `timeperiod_ranes`.

**Comment by nickchappell** [comment]
Hi @smbambling, 

This PR looks good, I just make inline comments in the diffs about the parameter names. To summarize, I'd like to keep the `timeperiod_` string off of the parameter names that aren't too specific to this defined type, like the `display_name` and `target_dir` and `_file` params. That way, this defined type would match up more closely with the existing defined types.

File: README.md (hunk 1 above)
Comment (smbambling) [inline README.md]: This is fixed and params now use timeperiod_X

**Comment by smbambling** [comment]
Updated to remove timeperiod_ from params as requested and README updated

**Comment by nickchappell** [comment]
Merged this in. Thanks @smbambling!

**Pull request closed without merging**